bin/check-instance-events.rb

```diff
@@ -66,55 +66,68 @@ def aws_config
       region: config[:aws_region] }
   end
 
+  def ec2_regions
+    # This is for SDK v2
+    # Aws.partition('aws').regions.map(&:name)
+
+    AWS::EC2.regions.map(&:name)
+  end
+
   def run
     event_instances = []
     aws_config = {}
 
+    aws_regions = ec2_regions
+
+    unless config[:aws_region].casecmp('all') == 0
```

Comment:
I recently added support for AWS_REGION environment variable supported by AWS api this logic will break that as AWS_REGION is used when -r is specified as an empty string.